### DISCUSSION TITLE
Nested routes at site root cause 404 error with actix-web

### DISCUSSION THREAD

**Discussion by bioinformatist** [leptos-rs/leptos#2030] [Q&A] [answered]
Hi dear Greg, sorry for reaching out again:confounded:

In a specific scenario I want to use nested routes at site root. However if my website was built in SSR mode, the error occurs:
 
![image](https://github.com/leptos-rs/leptos/assets/17194719/eeae8e4c-a848-4082-89bb-14d0c1b222e8)

To reproduce it, you may utilize the repo's `examples/todo_app_sqlite`, and modify the component `TodoApp()` as follows:

```rust
#[component]
pub fn TodoApp() -> impl IntoView {
    provide_meta_context();
    view! {

        <Link rel="shortcut icon" type_="image/ico" href="/favicon.ico"/>
        <Stylesheet id="leptos" href="/pkg/todo_app_sqlite.css"/>
        <Router>
            <header>
                <h1>"My Tasks"</h1>
            </header>
            <main>
                <Routes>
                    <Route path="" view=move || {
                        // Always show the child at last
                        view! { <Todos/><Outlet/>}
                    }>
                        <Route path=":id" view=move || {view! { <p>iiii</p> }}/>
                    </Route>
                </Routes>
            </main>
        </Router>
    }
}
```

Then navigating to the site root will result in a `404` error.

The similar nested routes in the `router` example (CSR mode) appears to work well.

I've observed that the error message `Unable to render directory without index file` is defined by [actix-web](https://github.com/actix/actix-web/blob/master/actix-files/src/error.rs#L13). However, it's difficult for me to determine the underlying cause and I'm uncertain if there's any misunderstanding. Thanks!

**Comment by gbj** [comment, marked as answer]
You do not have a fallback Route path="" at the nested level so `/:id` is the only complete path it will match 

See a bit more here https://leptos-rs.github.io/leptos/router/17_nested_routing.html#nested-routes-as-layout

**Reply by bioinformatist** [reply]
Yeah it's my misunderstanding here. It works well after the fallback route added. Thank you!
